args/core/containers/sparse_map.hpp

```diff
@@ -154,8 +160,64 @@ namespace args::core
 		{
 			return m_sparse.count(key) && m_sparse.at(key) >= 0 && m_sparse.at(key) < m_size && m_dense_key[m_sparse.at(key)] == key;
 		}
+
+		/**@brief Checks if all keys in sparse_map are inside this map as well.
+		 * @param other Other sparse_map to check against.
+		 * @returns bool True if all keys in other are also in this sparse_map, otherwise false.
+		 */
+		template<typename T>
+		A_NODISCARD bool contains(const sparse_map<key_type, T>& other) const
+		{
+			if (other.m_size == 0)
+				return true;
+
+			if (m_size == 0 || m_size < other.m_size)
+				return false;
+
+			bool overlap = true;
+			for (int i = 0; i < other.m_size; i++)
+				overlap = overlap && contains(other.m_dense_key[i]);
+
+			return overlap;
+		}
 #pragma endregion
 
+		/**@brief Checks if all keys and values and relations between them are the same for both sparse_maps.
+		 * @param other Other sparse_map to check against.
+		 * @returns bool True if both maps are the same size, contain the same keys, and all keys refer to the same value, otherwise false.
+		 */
+		A_NODISCARD bool equals(self_const_reference other) const
+		{
+			if (m_size == other.m_size)
+			{
+				bool equal = true;
+				for (int i = 0; i < m_size; i++)
+					equal = equal && other.contains(m_dense_key[i]) && get(m_dense_key[i]) == other.get(m_dense_key[i]);
+
+				return equal;
```

Comment:
there appears to be an early out option:
```suggestion
				for (int i = 0; i < m_size; ++i)
					if(!(other.contains(m_dense_key[i]) &&
					     get(m_dense_key[i]) == other.get(m_dense_key[i])))
						return false;
						
				return true;
```